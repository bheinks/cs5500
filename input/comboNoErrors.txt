(* 
   MIPL program with all kinds of things
   and no lexical errors
*)

program DiddleDeDum;
var stuff : array [1..20] of integer;
    data, max : integer;
    i, loc : integer;

  procedure sort;
  var i, n : integer;
      flag : boolean;

    procedure swap;
    var n : integer;

    begin (* swap *)
      n := stuff[i];
      stuff[i] := stuff[i+1];
      stuff[i+1] := n;
      flag := true
    end (* swap *)

  begin (* sort *)
    flag := true;
    n := max;
    while (n > 1) and flag do
    begin
      flag := false;
      i := 1;
      while i < n do
      begin
        if stuff[i] > stuff[i+1] then swap;
        i := i + 1
      end;

    end
  end (* sort *)

procedure find;
  var i, j, k : integer;
      found : boolean;

  begin (* find *)
    i := 1;
    j := max;
    found := false;
    while (i <= j) and not found do
    begin
      k := (i+j) div 2;
      if stuff[k] > data 
        then j := k - 1
        else if stuff[k] < data 
               then i := k + 1
               else found := true
    end;

    if found then loc := k else loc := 0
  end (* find *)

 begin (* DiddleDeDum *)
  max := 0;
  read(data);
  while data >= 0 do
  begin
    max := max + 1;
    stuff[max] := data;
    read(data)
  end;

  sort;
  write('(', ' ');
  i := 1;
  while i <= max do
  begin
    write(stuff[i], ' ');
    i := i + 1
  end;

  write(')', '\');
  read(data);
  while data >= 0 do
  begin
    find;
    if loc = 0
      then write('N', 'o', ' ', 'm', 'a', 't', 'c', 'h',
		 ' ', 'f', 'o', 'r', ' ', data, '\')
      else write('S', 't', 'u', 'f', 'f', '[', loc, ']',
		 ' ', '=', ' ', data, '\');
    read(data)
  end;

  write('e', 'n', 'd', '\');
 end (*DiddleDeDum *) .
