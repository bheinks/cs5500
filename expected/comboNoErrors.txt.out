TOKEN: T_PROG      LEXEME: program
TOKEN: T_IDENT     LEXEME: DiddleDeDum
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_VAR       LEXEME: var
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_COLON     LEXEME: :
TOKEN: T_ARRAY     LEXEME: array
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_DOTDOT    LEXEME: ..
TOKEN: T_INTCONST  LEXEME: 20
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_OF        LEXEME: of
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_COLON     LEXEME: :
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: loc
TOKEN: T_COLON     LEXEME: :
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_PROC      LEXEME: procedure
TOKEN: T_IDENT     LEXEME: sort
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_VAR       LEXEME: var
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_COLON     LEXEME: :
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: flag
TOKEN: T_COLON     LEXEME: :
TOKEN: T_BOOL      LEXEME: boolean
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_PROC      LEXEME: procedure
TOKEN: T_IDENT     LEXEME: swap
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_VAR       LEXEME: var
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_COLON     LEXEME: :
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: flag
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_TRUE      LEXEME: true
TOKEN: T_END       LEXEME: end
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: flag
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_TRUE      LEXEME: true
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_GT        LEXEME: >
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_AND       LEXEME: and
TOKEN: T_IDENT     LEXEME: flag
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: flag
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_FALSE     LEXEME: false
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_LT        LEXEME: <
TOKEN: T_IDENT     LEXEME: n
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IF        LEXEME: if
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_GT        LEXEME: >
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_THEN      LEXEME: then
TOKEN: T_IDENT     LEXEME: swap
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_END       LEXEME: end
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_END       LEXEME: end
TOKEN: T_END       LEXEME: end
TOKEN: T_PROC      LEXEME: procedure
TOKEN: T_IDENT     LEXEME: find
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_VAR       LEXEME: var
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: j
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_COLON     LEXEME: :
TOKEN: T_INT       LEXEME: integer
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: found
TOKEN: T_COLON     LEXEME: :
TOKEN: T_BOOL      LEXEME: boolean
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: j
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: found
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_FALSE     LEXEME: false
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_LE        LEXEME: <=
TOKEN: T_IDENT     LEXEME: j
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_AND       LEXEME: and
TOKEN: T_NOT       LEXEME: not
TOKEN: T_IDENT     LEXEME: found
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_IDENT     LEXEME: j
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_DIV       LEXEME: div
TOKEN: T_INTCONST  LEXEME: 2
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IF        LEXEME: if
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_GT        LEXEME: >
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_THEN      LEXEME: then
TOKEN: T_IDENT     LEXEME: j
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_MINUS     LEXEME: -
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_ELSE      LEXEME: else
TOKEN: T_IF        LEXEME: if
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_LT        LEXEME: <
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_THEN      LEXEME: then
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_ELSE      LEXEME: else
TOKEN: T_IDENT     LEXEME: found
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_TRUE      LEXEME: true
TOKEN: T_END       LEXEME: end
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IF        LEXEME: if
TOKEN: T_IDENT     LEXEME: found
TOKEN: T_THEN      LEXEME: then
TOKEN: T_IDENT     LEXEME: loc
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: k
TOKEN: T_ELSE      LEXEME: else
TOKEN: T_IDENT     LEXEME: loc
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_INTCONST  LEXEME: 0
TOKEN: T_END       LEXEME: end
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_INTCONST  LEXEME: 0
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_READ      LEXEME: read
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_GE        LEXEME: >=
TOKEN: T_INTCONST  LEXEME: 0
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_READ      LEXEME: read
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_END       LEXEME: end
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: sort
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_CHARCONST LEXEME: '('
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_LE        LEXEME: <=
TOKEN: T_IDENT     LEXEME: max
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: stuff
TOKEN: T_LBRACK    LEXEME: [
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_RBRACK    LEXEME: ]
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_ASSIGN    LEXEME: :=
TOKEN: T_IDENT     LEXEME: i
TOKEN: T_PLUS      LEXEME: +
TOKEN: T_INTCONST  LEXEME: 1
TOKEN: T_END       LEXEME: end
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_CHARCONST LEXEME: ')'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '\'
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_READ      LEXEME: read
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WHILE     LEXEME: while
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_GE        LEXEME: >=
TOKEN: T_INTCONST  LEXEME: 0
TOKEN: T_DO        LEXEME: do
TOKEN: T_BEGIN     LEXEME: begin
TOKEN: T_IDENT     LEXEME: find
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_IF        LEXEME: if
TOKEN: T_IDENT     LEXEME: loc
TOKEN: T_EQ        LEXEME: =
TOKEN: T_INTCONST  LEXEME: 0
TOKEN: T_THEN      LEXEME: then
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_CHARCONST LEXEME: 'N'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'o'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'm'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'a'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 't'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'c'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'h'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'f'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'o'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'r'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '\'
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_ELSE      LEXEME: else
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_CHARCONST LEXEME: 'S'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 't'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'u'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'f'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'f'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '['
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: loc
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ']'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '='
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: ' '
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '\'
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_READ      LEXEME: read
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_IDENT     LEXEME: data
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_END       LEXEME: end
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_WRITE     LEXEME: write
TOKEN: T_LPAREN    LEXEME: (
TOKEN: T_CHARCONST LEXEME: 'e'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'n'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: 'd'
TOKEN: T_COMMA     LEXEME: ,
TOKEN: T_CHARCONST LEXEME: '\'
TOKEN: T_RPAREN    LEXEME: )
TOKEN: T_SCOLON    LEXEME: ;
TOKEN: T_END       LEXEME: end
TOKEN: T_DOT       LEXEME: .
